web/src/features/time/TimeAxis.tsx

```diff
@@ -3,17 +3,25 @@ import { useTranslation } from 'react-i18next';
 import PulseLoader from 'react-spinners/PulseLoader';
 import useResizeObserver from 'use-resize-observer/polyfilled';
 import { HOURLY_TIME_INDEX, TimeRange } from 'utils/constants';
-import { isValidHistoricalTimeRange } from 'utils/helpers';
+import { getLocalTime, isValidHistoricalTimeRange } from 'utils/helpers';
 
 import { formatDateTick } from '../../utils/formatting';
 
-// Frequency at which values are displayed for a tick
-const TIME_TO_TICK_FREQUENCY = {
-  '24h': 6,
-  '72h': 12,
-  '30d': 6,
-  '12mo': 1,
-  all: 1,
+// The following represents a list of methods, indexed by time range, that depict
+// if a datetime should be a major tick, where we will display the date value.
+type MajorTickCallable = (
+  _localHours: number,
+  _localMinutes: number,
+  _index: number
+) => boolean;
+type IsMajorTickCallableType = { [key: string]: MajorTickCallable };
+const IS_MAJOR_TICK_CALLABLE: IsMajorTickCallableType = {
+  '24h': (_localHours, _localMinutes, _index) => _index % 6 === 0,
+  '72h': (_localHours, _localMinutes, _index) =>
+    (_localHours === 12 || _localHours === 0) && _localMinutes === 0,
+  '30d': (_localHours, _localMinutes, _index) => _index % 6 === 0,
+  '12mo': (_localHours, _localMinutes, _index) => true,
+  all: (_localHours, _localMinutes, _index) => true,
 };
```

Comment:
This made my brain hurt, what about doing this instead:
```
const isMajorTick = (_localHours, _localMinutes, _index) => {
  switch (key) {
    case '24h':
    case '30d':
      return _index % 6 === 0;
    case '72h':
      return (_localHours === 12 || _localHours === 0) && _localMinutes === 0;
    case '12mo':
    case 'all':
      return true;
    default:
      return false;
  }
};
```
Some other changes are needed but in general this is more easy to understand I think.

I'd be happy to make them.

Reply:
Makes sense!